cmd/cmd.go

```diff
@@ -90,7 +91,7 @@ func ExecInCoreSnap() {
 	// ensure we do not re-exec into an older version of snapd, look
 	// for info file and ignore version of core that do not yet have
 	// it
-	fullInfo := filepath.Join(corePath, "/usr/lib/snapd/info")
```

Comment:
@zyga Please note that the code now uses dirs.CoreLibExecDir which will expand to /usr/lib/snapd even on Fedora. A previous version of the code was using dirs.DistroLibExecDir which is, as you pointed out, wrong.

In summary: The new code `filepath.Join(corePath, dirs.CoreLibExecDir, "info")` will take the `corePath` as determined before through `/proc/self/exe` and append dirs.CoreLibExecDir so that we end up with `/snap/core/current/usr/libexec/snapd/info` on Ubuntu and `/var/lib/snapd/snap/core/current/usr/libexec/snapd/info`